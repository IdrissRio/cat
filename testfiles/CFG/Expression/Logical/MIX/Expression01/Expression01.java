public class Expression01 {
  void m(int a, int b, int c, int d) {
    boolean r;
    r = !(a > 0) && !(b > 0) || c > 0 && d > 0;
  }
}