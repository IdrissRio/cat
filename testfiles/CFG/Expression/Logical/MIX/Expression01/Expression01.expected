NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [VariableDeclarator[59]]
NodeName: VariableDeclarator[59] Successors: [VarAccess[70]]
NodeName: VarAccess[70] Successors: [IntegerLiteral[71]]
NodeName: IntegerLiteral[71] Successors: [GTExpr[69]]
NodeName: GTExpr[69] Successors: [LogNotExpr[67]]
NodeName: LogNotExpr[67] Successors: [VarAccess[75], VarAccess[79]]
NodeName: VarAccess[75] Successors: [IntegerLiteral[76]]
NodeName: IntegerLiteral[76] Successors: [GTExpr[74]]
NodeName: GTExpr[74] Successors: [LogNotExpr[72]]
NodeName: LogNotExpr[72] Successors: [AssignSimpleExpr[63], VarAccess[79]]
NodeName: AssignSimpleExpr[63] Successors: [Exit[52]]
NodeName: Exit[52] Successors: []
NodeName: VarAccess[79] Successors: [IntegerLiteral[80]]
NodeName: IntegerLiteral[80] Successors: [GTExpr[78]]
NodeName: GTExpr[78] Successors: [VarAccess[82], AssignSimpleExpr[63]]
NodeName: VarAccess[82] Successors: [IntegerLiteral[83]]
NodeName: IntegerLiteral[83] Successors: [GTExpr[81]]
NodeName: GTExpr[81] Successors: [AssignSimpleExpr[63]]

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[52] Predecessor: [AssignSimpleExpr[63]]
NodeName: AssignSimpleExpr[63] Predecessor: [LogNotExpr[72], GTExpr[78], GTExpr[81]]
NodeName: LogNotExpr[72] Predecessor: [GTExpr[74]]
NodeName: GTExpr[74] Predecessor: [IntegerLiteral[76]]
NodeName: IntegerLiteral[76] Predecessor: [VarAccess[75]]
NodeName: VarAccess[75] Predecessor: [LogNotExpr[67]]
NodeName: LogNotExpr[67] Predecessor: [GTExpr[69]]
NodeName: GTExpr[69] Predecessor: [IntegerLiteral[71]]
NodeName: IntegerLiteral[71] Predecessor: [VarAccess[70]]
NodeName: VarAccess[70] Predecessor: [VariableDeclarator[59]]
NodeName: VariableDeclarator[59] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: GTExpr[78] Predecessor: [IntegerLiteral[80]]
NodeName: IntegerLiteral[80] Predecessor: [VarAccess[79]]
NodeName: VarAccess[79] Predecessor: [LogNotExpr[67], LogNotExpr[72]]
NodeName: GTExpr[81] Predecessor: [IntegerLiteral[83]]
NodeName: IntegerLiteral[83] Predecessor: [VarAccess[82]]
NodeName: VarAccess[82] Predecessor: [GTExpr[78]]
